src/ajax.js

```diff
@@ -444,80 +444,80 @@ jQuery.extend( {
 			// Default abort message
 			strAbort = "canceled",
 
-			// Fake xhr
-			jqXHR = {
-				readyState: 0,
+			// Fake xhr (ref the true Deferred instance above)
+			jqXHR = deferred;
```

Comment:
Could you elaborate through when can this **circumvent the normal flow** happen in the browser please? `.resolve` is a perfectly fine Promise API, why would someone call `.resolve` before XMLHttpRequest resolve it for him/her? Assuming the var is always on the call stack (scoped) not globally hackable through a js console.